docs/running-tests/wptreport-schema.json

```diff
@@ -0,0 +1,312 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "WptReport",
+  "type": "object",
+  "required": [
+    "results",
+    "run_info",
+    "time_end",
+    "time_start"
+  ],
+  "properties": {
+    "lsan_leaks": {
+      "type": [
+        "array",
+        "null"
+      ],
+      "items": {
+        "$ref": "#/definitions/LsanLeak"
+      }
+    },
+    "mozleak": {
+      "type": [
+        "object",
+        "null"
+      ],
+      "additionalProperties": {
+        "$ref": "#/definitions/MozLeak"
+      }
+    },
+    "results": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/TestResult"
+      }
+    },
+    "run_info": true,
```

Comment:
@jgraham do we want to give some guarantees as to what appears in `run_info`?

Per https://github.com/web-platform-tests/wpt/blob/7f6dc72ad0f109d545c2068e0b336dd9af0b3e0a/tools/wptrunner/wptrunner/wpttest.py#L83, the following always exist: `python_version`, `product`, `verify`, `headless`, and `webrender`(!!!!).

That said, looking through those I guess none of them are actually that useful in general… (product is, but it's of limited use without product version and OS)

Reply:
No, I don't think we do. It's kind of annoying, but I think we want to allow people to pass back empty `run_info` if they aren't comparing to other products/configurations. In practice specific expectation metadata does depend on specific properties exising, but the metadata isn't really terribly portable beyond the system it runs in, so that isn't obviously a sufficient justification for standardising this.

Reply:
Not even just making it a nullable object?

Reply:
I think the guarantee for RunInfo is that it's a `Map<String, JsonValue>` (in some Rustish syntax)`. I don't think it should be null but I don't think we can guarantee any specific fields.